plugin/completion/flags_manager.py

```diff
@@ -298,10 +299,9 @@ def write_flags_to_file(self, new_flags, file_path, strategy):
             if flag_strategy == "keep_old":
                 return
             if flag_strategy == "merge":
-                # union of two flags sets
-                curr_flags = set(self.flags_from_clang_file(
-                    File(file_path)))
-                new_flags = new_flags.union(curr_flags)
+                # union of two flag arrays
+                curr_flags = self.flags_from_clang_file(File(file_path))
+                new_flags = curr_flags + list(set(new_flags) - set(curr_flags))
```

Comment:
Yes, it will put them in the end of the list but in an unspecified order as the result of `set(new_flags) - set(curr_flags)` is itself a `set`, so the order of this difference will be arbitrary. 

And yes, currently it doesn't really matter as what you say is totally correct and after parsing cmake the damage is already done.